homeassistant/components/zwave_js/update.py

```diff
@@ -110,10 +110,9 @@ def _update_on_status_change(self, _: dict[str, Any]) -> None:
     def _update_progress(self, event: dict[str, Any]) -> None:
         """Update install progress on event."""
         progress: FirmwareUpdateProgress = event["firmware_update_progress"]
-        if not self._latest_version_firmware:
-            return
-        self._attr_in_progress = int(progress.progress)
-        self.async_write_ha_state()
+        if self._latest_version_firmware:
```

Comment:
I don't think we should make the code harder to read to fake higher coverage. If there's more than one line indented I prefer guard clause.

Reply:
fair enough, I don't know how to trigger this though, not sure if this guard is actually needed.

Reply:
If it can't happen we can remove it. Let's do that separately though, in that case.